### PR TITLE
feat: add postal read consumer

### PR DESCRIPTION
**Pull request by pgallik**

### CODE REVIEW

File: src/StreetNameRegistry.Consumer.Read.Postal/appsettings.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
+  "ConnectionStrings": {
+    "ConsumerPostal": "Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory.StreetNameRegistry;Trusted_Connection=True;TrustServerCertificate=True;",
+    "ConsumerPostalAdmin": "Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory.StreetNameRegistry;Trusted_Connection=True;TrustServerCertificate=True;"
+  },
+
+  "DataDog": {
+    "Enabled": false,
+    "Debug": false,
+    "ServiceName": "streetname-registry-consumer-read-Postal-dev"
```
Comment (ArneD) [inline src/StreetNameRegistry.Consumer.Read.Postal/appsettings.json]: lowercase `P`

File: src/StreetNameRegistry.Consumer.Read.Postal/init.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+CONTAINERID=$(curl -s http://169.254.170.2/v2/metadata | jq -r ".Containers[] | select(.Labels[\"com.amazonaws.ecs.container-name\"] | startswith(\"basisregisters-\") and endswith(\"-consumer-read-Postal\")) | .DockerId")
```
Comment (ArneD) [inline src/StreetNameRegistry.Consumer.Read.Postal/init.sh]: lowercase `Postal`

File: src/StreetNameRegistry.Consumer.Read.Postal/StreetNameRegistry.Consumer.Read.Postal.csproj [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    <ProjectReference Include="..\StreetNameRegistry.Infrastructure\StreetNameRegistry.Infrastructure.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Compile Include="..\..\SolutionInfo.cs" Link="Properties\SolutionInfo.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Folder Include="Migrations" />
+  </ItemGroup>
```
Comment (ArneD) [inline src/StreetNameRegistry.Consumer.Read.Postal/StreetNameRegistry.Consumer.Read.Postal.csproj]: Not needed

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -248,6 +248,21 @@ jobs:
       semver: ${{ needs.set-release-version.outputs.version }}
     secrets: inherit
 
+  build-consumer-read-postal:
+    name: Build Snapshot Verifier
```
Comment (jvandaal) [inline .github/workflows/release.yml]: Name is still Snapshot Verifier

File: src/StreetNameRegistry.Api.Legacy/StreetName/Query/StreetNameListQueryV2.cs [hunk 5]
```diff
@@ -17,13 +18,15 @@ public sealed class StreetNameListQueryV2 : Query<StreetNameListItemV2, StreetNa
     {
         private readonly LegacyContext _legacyContext;
         private readonly SyndicationContext _syndicationContext;
+        private readonly ConsumerPostalContext _PostalContext;
```
Comment (ArneD) [inline src/StreetNameRegistry.Api.Legacy/StreetName/Query/StreetNameListQueryV2.cs]: lowercase `P`

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -248,6 +248,21 @@ jobs:
...
+    uses: Informatievlaanderen/build-pipeline/.github/workflows/build-image.yml@main
+    needs: [ set-release-version ]
+    if: ${{ (github.repository_owner == 'Informatievlaanderen') && (needs.set-release-version.outputs.version != 'none') }}
+    with:
+      registry: streetname-registry
+      build-target: Containerize_ConsumerReadPostal
+      image-file: sr-consumer-read-postal-image.tar
+      image-name: consumer-read-postal
+      test-project: StreetNameRegistry.Tests
+      build-project: StreetNameRegistry.Snapshot.Verifier
```
Comment (jvandaal) [inline .github/workflows/release.yml]: Project is also still targetting Snapshot Verifier

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/StreetNameRegistry.Api.Legacy/StreetName/Query/StreetNameListQueryV2.cs [hunk 7]
```diff
@@ -17,13 +18,15 @@ public sealed class StreetNameListQueryV2 : Query<StreetNameListItemV2, StreetNa
...
 
         protected override ISorting Sorting => new StreetNameSorting();
 
-        public StreetNameListQueryV2(LegacyContext legacyContext, SyndicationContext syndicationContext)
+        public StreetNameListQueryV2(LegacyContext legacyContext, SyndicationContext syndicationContext, ConsumerPostalContext postalContext)
         {
             _legacyContext = legacyContext;
             _syndicationContext = syndicationContext;
+            _PostalContext = postalContext;
```
Comment (jvandaal) [inline src/StreetNameRegistry.Api.Legacy/StreetName/Query/StreetNameListQueryV2.cs]: small caps _P

File: src/StreetNameRegistry.Api.Legacy/StreetName/Query/StreetNameListQuery.cs [hunk 8]
```diff
@@ -89,6 +92,15 @@ protected override IQueryable<StreetNameListItem> Filter(FilteringHeader<StreetN
                 }
             }
 
+            if (filtering.Filter.PostalCode is not null)
```
Comment (jvandaal) [inline src/StreetNameRegistry.Api.Legacy/StreetName/Query/StreetNameListQuery.cs]: !string.IsNotNullOrWhitespace?

File: src/StreetNameRegistry.Api.Legacy/StreetName/Query/StreetNameListQueryV2.cs [hunk 9]
```diff
@@ -107,6 +110,15 @@ protected override IQueryable<StreetNameListItemV2> Filter(FilteringHeader<Stree
                 }
             }
 
+            if (filtering.Filter.PostalCode is not null)
```
Comment (jvandaal) [inline src/StreetNameRegistry.Api.Legacy/StreetName/Query/StreetNameListQueryV2.cs]: !string.IsNotNullOrWhitespace?

**Review by emalfroy — DISMISSED** [review]

File: .github/workflows/release.yml [hunk 10]
```diff
@@ -979,6 +999,7 @@ jobs:
           , 'streetname-registry-producer-snapshot-oslo'
           , 'streetname-registry-projections-backoffice'
           , 'streetname-registry-snapshot-verifier'
+          , 'streetname-registry-consumer-read-postal'
```
Comment (ArneD) [inline .github/workflows/release.yml]: remove from deploy, until it's configured by AWS

File: src/StreetNameRegistry.Api.Legacy/StreetName/List/ListHandler.cs [hunk 11]
```diff
@@ -19,21 +20,24 @@ public sealed class ListHandler : ListHandlerBase
     {
         private readonly LegacyContext _legacyContext;
         private readonly SyndicationContext _syndicationContext;
+        private readonly ConsumerPostalContext _PostalContext;
```
Comment (ArneD) [inline src/StreetNameRegistry.Api.Legacy/StreetName/List/ListHandler.cs]: lowercase `P`

File: src/StreetNameRegistry.Api.Legacy/StreetName/Query/StreetNameListQuery.cs [hunk 12]
```diff
@@ -89,6 +92,15 @@ protected override IQueryable<StreetNameListItem> Filter(FilteringHeader<StreetN
                 }
             }
 
+            if (!string.IsNullOrWhiteSpace(filtering.Filter.PostalCode))
+            {
+                var postalConsumerItem = _postalContext.PostalConsumerItems.Find(filtering.Filter.PostalCode);
+                if (postalConsumerItem?.NisCode != null)
+                {
+                    streetNames = streetNames.Where(m => m.NisCode == postalConsumerItem.NisCode);
+                }
```
Comment (ArneD) [inline src/StreetNameRegistry.Api.Legacy/StreetName/Query/StreetNameListQuery.cs:101]:
```
else 
{
// return empty
}
```

File: src/StreetNameRegistry.Api.Legacy/StreetName/Query/StreetNameListQueryV2.cs [hunk 13]
```diff
@@ -107,6 +110,16 @@ protected override IQueryable<StreetNameListItemV2> Filter(FilteringHeader<Stree
...
             }
 
+            if (!string.IsNullOrWhiteSpace(filtering.Filter.PostalCode))
+            {
+                var postalConsumerItem = _postalContext.PostalConsumerItems.Find(filtering.Filter.PostalCode);
+
+                if (postalConsumerItem?.NisCode != null)
+                {
+                    streetNames = streetNames.Where(m => m.NisCode == postalConsumerItem.NisCode);
+                }
```
Comment (ArneD) [inline src/StreetNameRegistry.Api.Legacy/StreetName/Query/StreetNameListQueryV2.cs:120]: else here too

File: src/StreetNameRegistry.Api.Oslo/StreetName/Query/StreetNameListOsloQuery.cs [hunk 14]
```diff
@@ -94,6 +100,15 @@ protected override IQueryable<StreetNameListItem> Filter(FilteringHeader<StreetN
                     streetNames = streetNames.Where(m => m.Status.HasValue && (int)m.Status.Value == -1);
             }
 
+            if (!string.IsNullOrWhiteSpace(filtering.Filter.PostalCode))
+            {
+                var postalConsumerItem = _postalContext.PostalConsumerItems.Find(filtering.Filter.PostalCode);
+                if (postalConsumerItem?.NisCode != null)
+                {
+                    streetNames = streetNames.Where(m => m.NisCode == postalConsumerItem.NisCode);
+                }
```
Comment (ArneD) [inline src/StreetNameRegistry.Api.Oslo/StreetName/Query/StreetNameListOsloQuery.cs:109]: else

File: src/StreetNameRegistry.Api.Oslo/StreetName/Query/StreetNameListOsloQueryV2.cs [hunk 15]
```diff
@@ -114,6 +120,15 @@ protected override IQueryable<StreetNameListItemV2> Filter(FilteringHeader<Stree
                 }
             }
 
+            if (!string.IsNullOrWhiteSpace(filtering.Filter.PostalCode))
+            {
+                var postalConsumerItem = _postalContext.PostalConsumerItems.Find(filtering.Filter.PostalCode);
+                if (postalConsumerItem?.NisCode != null)
+                {
+                    streetNames = streetNames.Where(m => m.NisCode == postalConsumerItem.NisCode);
+                }
```
Comment (ArneD) [inline src/StreetNameRegistry.Api.Oslo/StreetName/Query/StreetNameListOsloQueryV2.cs:129]: else

File: src/StreetNameRegistry.Consumer.Read.Postal/PostalConsumerItem.cs [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+
+    public class PostalConsumerItem
+    {
+        public string PostalCode { get; set; }
+        public string? NisCode { get; set; }
+        public PostalStatus Status { get; set; }
+        public string? NameDutch { get; set; }
+        public string? NameFrench { get; set; }
+        public string? NameGerman { get; set; }
+        public string? NameEnglish { get; set; }
```
Comment (ArneD) [inline src/StreetNameRegistry.Consumer.Read.Postal/PostalConsumerItem.cs]: What with multiple names?
https://api.prd.basisregisters.vlaanderen.be/v1/postinfo/8200

**Review by ArneD — CHANGES_REQUESTED** [review]

**Review by ArneD — CHANGES_REQUESTED** [review]
still in deploy steps

**Review by ArneD — APPROVED** [review]

**Pull request merged**